objects/src/notes/mod.rs

```diff
@@ -34,206 +34,22 @@ pub use nullifier::Nullifier;
 mod origin;
 pub use origin::{NoteInclusionProof, NoteOrigin};
 
+mod output;
+pub use output::NoteOutput;
+
+mod recipient;
+pub use recipient::NoteRecipient;
+
 mod script;
 pub use script::NoteScript;
 
+mod vault;
+pub use vault::NoteVault;
+
 // CONSTANTS
 // ================================================================================================
 
 /// The depth of the leafs in the note Merkle tree used to commit to notes produced in a block.
 /// This is equal `NOTE_TREE_DEPTH + 1`. In the kernel we do not authenticate leaf data directly
 /// but rather authenticate hash(left_leaf, right_leaf).
 pub const NOTE_LEAF_DEPTH: u8 = NOTE_TREE_DEPTH + 1;
-
-// NOTE
-// ================================================================================================
-
-/// A note which can be used to transfer assets between accounts.
-///
-/// This struct is a full description of a note which is needed to execute a note in a transaction.
-/// A note consists of:
-///
-/// Core on-chain data which is used to execute a note:
-/// - A script which must be executed in a context of some account to claim the assets.
-/// - A set of inputs which can be read to memory during script execution via the invocation of the
-///   `note::get_inputs` in the kernel API.
-/// - A set of assets stored in the note.
-/// - A serial number which can be used to break linkability between note hash and note nullifier.
-///
-/// Auxiliary data which is used to verify authenticity and signal additional information:
-/// - A metadata object which contains information about the sender, the tag and the number of
-///   assets in the note.
-#[derive(Clone, Debug, PartialEq, Eq)]
-pub struct Note {
-    script: NoteScript,
-    inputs: NoteInputs,
-    assets: NoteAssets,
-    serial_num: Word,
-    metadata: NoteMetadata,
-
-    id: NoteId,
-    recipient: Digest,
-    nullifier: Nullifier,
-}
-
-fn compute_recipient(serial_num: Word, script: &NoteScript, inputs: &NoteInputs) -> Digest {
-    let serial_num_hash = Hasher::merge(&[serial_num.into(), Digest::default()]);
-    let merge_script = Hasher::merge(&[serial_num_hash, script.hash()]);
-    Hasher::merge(&[merge_script, inputs.commitment()])
-}
-
-impl Note {
```

Comment:
Renamed to `NoteConsumable` and moved to the `consumable` module. For our current usage, `Note` is too generic, the new name should make a bit cleared that this structure is used to hold the details necessary to consume a note as an input.